spec/app/domain/services/create_course/service_spec.rb

```diff
@@ -0,0 +1,82 @@
+require 'rails_helper'
+
+RSpec.describe Services::CreateCourse::Service do
+  let(:service) { Services::CreateCourse::Service.new }
+  let(:action) { process }
+
+  def process
+    service.process(course_uuid: given_course_uuid, ecosystem_uuid: given_ecosystem_uuid)
+  end
+
+  context "when non-existing ecosystem is given" do
+    let(:given_course_uuid) { SecureRandom.uuid.to_s }
+    let(:given_ecosystem_uuid) { SecureRandom.uuid.to_s }
+
+    let!(:split_time) { time = Time.now; sleep(0.001); time }
+
+    it "raises error" do
+      expect{action}.to raise_error(Errors::AppUnprocessableError)
+    end
+  end
+
+  context "when existing ecosystem is given" do
+
+    ecosystem_uuid = SecureRandom.uuid.to_s
+    course_uuid = SecureRandom.uuid.to_s
+
+    let(:given_course_uuid) { course_uuid }
+    let(:given_ecosystem_uuid) { ecosystem_uuid }
+
+    let!(:split_time) { time = Time.now; sleep(0.001); time }
+
+    expected = {created_course_uuid: course_uuid}
+
+    before(:each) do
+      Ecosystem.find_or_create_by(uuid: ecosystem_uuid)
+    end
+
+    it "Course is created" do
+      expect{action}.to change{Course.count}
+    end
+
+    it "created_course_uuid is returned" do
+      expect(action).to eq expected
```

Comment:
I think it's easier to word, debug, and  grok:
```ruby
it "the Course's uuid is returned" do
  expect(action.fetch(:created_course_uuid)).to eq(given_course_uuid)
end
```
(Using `fetch` will raise an error if the given key is missing.)